test/Generics/protocol_requirement_signatures.swift

```diff
@@ -0,0 +1,60 @@
+// RUN: %target-typecheck-verify-swift -typecheck %s -verify
+// RUN: %target-typecheck-verify-swift -typecheck -debug-generic-signatures %s > %t.dump 2>&1 
+// RUN: %FileCheck %s < %t.dump
+
+// CHECK-LABEL: .P1@
+// CHECK-NEXT: Requirements:
+// CHECK-NEXT:   Self : P1 [protocol_self]
+// CHECK-NEXT: Generic signature: <Self where Self : P1>
+// CHECK-NEXT: Canonical generic signature: <τ_0_0 where τ_0_0 : P1>
```

Comment:
We can ask clients to ignore it, I guess. It just feels wrong to have this requirement in the signature because it makes all of the other requirements redundant.